docker_build.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+# Purpose : Build ruby using docker image ubuntu:18.04
+# Author  : Ky-Anh Huynh
+# License : MIT
+# Date    : 2019-Feb-16
+
+build_docker_image() {
+  {
+    : "${DOCKER_IMAGE:=ubuntu:16.04}"
+    echo "FROM ${DOCKER_IMAGE}"
+    echo "ENV DOCKER_IMAGE ${DOCKER_IMAGE}"
+    echo "ADD ./docker_build.sh /root/"
+    echo "RUN bash /root/docker_build.sh __build_base"
+  } > Dockerfile.rubyc_build_base
+
+  docker build -f Dockerfile.rubyc_build_base -t "rubyc_build_base" .
+}
+
+__setup() {
+  set -uex
+
+  echo "## Your build environment"
+  env
+  echo "## DNS resolvers"
+  cat /etc/resolv.conf
+
+  export PATH="$HOME/.rbenv/plugins/ruby-build/bin:$HOME/.rbenv/bin:$PATH"
+  if command -v rbenv >/dev/null; then
+    eval "$(rbenv init -)"
+  fi
+}
+
+__build_base() {
+  __apt_get_install() {
+    apt-get update
+    echo Y | apt-get install \
+      build-essential \
+      bison git \
+      openssl \
+      libssl-dev \
+      texinfo \
+      squashfs-tools \
+      wget \
+      libreadline-dev zlib1g-dev \
+      pkg-config
+  }
+
+  __apt_get_install
+
+  [[ -d ~/.rbenv ]] \
+  || git clone https://github.com/rbenv/rbenv.git ~/.rbenv
+  [[ -d ~/.rbenv/plugins/ruby-build ]] \
+  || git clone https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build
+
+  __setup
+  rbenv install 2.6.0
```

Comment:
Probably want 2.6.1 as 2.6.0 has the [http security issue](https://mensfeld.pl/2019/01/exploring-a-critical-netprotocol-issue-in-ruby-2-6-0p0-and-how-it-can-lead-to-a-security-problem/).

But that would then require updating the ruby sources again and it was PITA and I'm not sure if it even worked right (we're not using the 2.6 branch because there were some problems).

Reply:
@kke  I have updated the PR to support any version. I'm using branch 2.6.0 so far it's good (even if there are some security issues, as we don't expose our app.) So far I spent quite a lot of time to fix the build process, and I hope this PR can help someone to save their time in case they want to try 2.6.0 for any purpose. 

I can also add 4.0 support (already add TODO item; I had the script locally but I have imported to this `docker_build.sh`). My primary purpose is to avoid travis CI/ and/or any CI-specific feature: I don't use it on my laptop, and I don't really like the idea of downloading ruby-4.0 from the plain HTTP site (http://enclose.io/, it's plain!) So far this is more critical than using Ruby-2.6.0 :D 

It's ok if the PR is rejected: the file is still on public domain . 

Thanks a lot for your works.